src/lib/zkapps_examples/big_circuit/zkapps_big_circuit.ml

```diff
@@ -0,0 +1,44 @@
+open Signature_lib
+open Snark_params.Tick.Run
+open Mina_base
+
+(** Constructs a circuit with around [num_constraints] constraints. 
+    Note that since the frontend of snarky doesn't know the number of constraints it creates, 
+    we can't create a circuit the exact number of constraints we want. We can only approximate.
+  *)
+let big_circuit num_constraints =
+  for i = 0 to num_constraints * 2 do
+    let a = constant Field.typ (Field.Constant.of_int i) in
+    let b = exists Field.typ ~compute:(fun () -> Field.Constant.of_int i) in
+    Field.Assert.equal a b
+  done
+
+(* Make sure that the helper works. *)
+let () =
+  let input_typ = Typ.unit in
+  let return_typ = Typ.unit in
+  let expected_constraints = 1 lsl 11 in
+  let circuit _ _ = big_circuit expected_constraints in
+  let cs = constraint_system ~input_typ ~return_typ circuit in
+  let num_constraints =
+    Kimchi_backend.Pasta.Vesta_based_plonk.R1CS_constraint_system
+    .num_constraints cs
+  in
+  let _witness = generate_witness ~input_typ ~return_typ circuit () in
+  if num_constraints < expected_constraints then (
+    Format.eprintf "constraints: %d, wanted: %d\n" num_constraints
```

Comment:
are you sure that it's `@.`? It says `!` in https://v2.ocaml.org/api/Printf.html

Reply:
<img width="717" alt="Screenshot 2023-01-23 at 12 57 18 PM" src="https://user-images.githubusercontent.com/1316043/214148367-37ab2a33-6295-4d2c-baa4-5d485695ba1c.png">
ok there's this in https://v2.ocaml.org/api/Format.html

Reply:
in any case it should work if you run your test with `--no-buffer` no?

Reply:
1. What is `--no-buffer` ?

Reply:
`--no-buffer` is to display these when you run tests (otherwise sometimes printing doesn't work)

Reply:
| in any case it should work if you run with --no-buffer no?

My understanding is that `--no-buffer` gives you direct access to the console (aka `dune` does not buffer the output). It does not follow that `Format.printf "foo"` will print out `foo` since `stdout` is not necessarily flushed (even if you have direct access to the terminal). `Format` still buffers its output -- it needs to in order to enforce box-based structured pretty-printing
.